connector/countconnector/counter.go

```diff
@@ -117,3 +125,24 @@ func (c *counter[K]) appendMetricsTo(metricSlice pmetric.MetricSlice) {
 		}
 	}
 }
+
+func getNestedAttribute(attr string, depth int, attributes pcommon.Map) (pcommon.Value, bool) {
+	logger := zap.NewExample()
+	defer logger.Sync()
+	left, right, _ := strings.Cut(attr, ".")
+	av, ok := attributes.Get(left)
+	if ok && depth > 0 {
```

Comment:
Let's pull the depth check up to the top of the function. I think if it's 0, we can just return immediately.